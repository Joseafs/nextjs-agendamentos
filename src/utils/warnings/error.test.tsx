import { render } from '@testing-library/react';
import {
  errorSchedulingConflicts,
  errorSchedulingEndBiggerThanStart,
  errorSchedulingTitle
} from './error';

describe('Warnings Error', () => {
  it('Should match text and count of errorSchedulingConflicts', () => {
    const { container } = render(errorSchedulingConflicts(5));

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent('Conflito(s) Identificado(s): 5');
  });
  it('Should match text of errorSchedulingEndBiggerThanStart', () => {
    const { container } = render(errorSchedulingEndBiggerThanStart());
    expect(container).toMatchSnapshot();
  });
  it('Should match text of errorSchedulingTitle', () => {
    const { container } = render(errorSchedulingTitle());
    expect(container).toMatchSnapshot();
  });
});
