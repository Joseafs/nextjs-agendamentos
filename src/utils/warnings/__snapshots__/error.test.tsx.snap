// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warnings Error Should match text and count of errorSchedulingConflicts 1`] = `
<div>
  <h5>
    Novo agendamento não pode ocupar o mesmo periodo de outro já cadastrado
  </h5>
  <h6>
    Conflito(s) Identificado(s): 
    <strong>
      5
    </strong>
  </h6>
</div>
`;

exports[`Warnings Error Should match text of errorSchedulingEndBiggerThanStart 1`] = `
<div>
  <h5>
    Data inserida no campo 'Dia e horário de fim' deve ser menor que o 'Dia e horário de início'
  </h5>
  <h6>
    Favor selecionar horarios compatíveis... (Será considerado a partir de 1min)
  </h6>
</div>
`;

exports[`Warnings Error Should match text of errorSchedulingTitle 1`] = `
<div>
  <h5>
    Título precisa ter no mínimo 3 caracteres
  </h5>
</div>
`;
